src/java/org/apache/cassandra/hints/ChecksummedDataInput.java

```diff
@@ -43,6 +44,8 @@
  */
 public class ChecksummedDataInput extends RebufferingInputStream
 {
+    private static final BufferPool bufferPool = BufferPoolManager.permanent();
```

Comment:
It's been used for a while now. We need a fixed size buffer here, and it makes sense to reuse an existing pool instead of implementing an ad-hoc one. That said, this particular use case would probably be just fine w/ allocating a buffer when needed instead of pooling at all.

Reply:
changed to allocating new buffer instead of pooled buffer.

Reply:
@jasonstack The only thing, having looked at this a bit more, is that `CompressedChecksummedDataInput` still uses the pool. Is it possible for `readBuffer()` to put the initial buffer, which wasn't taken from the pool, back into the pool?